test/parallel/test-child-process-spawnsync-validation-errors.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const spawnSync = require('child_process').spawnSync;
+
+function pass(option, value) {
+  // Run the command with the specified option. Since it's not a real command,
+  // spawnSync() should run successfully but return an ENOENT error.
+  const child = spawnSync('not_a_real_command', { [option]: value });
+
+  assert.strictEqual(child.error.code, 'ENOENT');
+}
+
+function fail(option, value, message) {
+  assert.throws(() => {
+    spawnSync('not_a_real_command', { [option]: value });
+  }, message);
+}
+
+{
+  // Validate the cwd option
+  const err = /TypeError: "cwd" must be a string/;
+
+  pass('cwd', undefined);
+  pass('cwd', null);
+  pass('cwd', __dirname);
+  fail('cwd', 0, err);
+  fail('cwd', 1, err);
+  fail('cwd', true, err);
+  fail('cwd', false, err);
+  fail('cwd', [], err);
+  fail('cwd', {}, err);
+}
+
+{
+  // Validate the detached option
+  const err = /TypeError: "detached" must be a boolean/;
+
+  pass('detached', undefined);
+  pass('detached', null);
+  pass('detached', true);
+  pass('detached', false);
+  fail('detached', 0, err);
+  fail('detached', 1, err);
+  fail('detached', __dirname, err);
+  fail('detached', [], err);
+  fail('detached', {}, err);
+}
+
+{
+  // Validate the uid option
+  if (!(common.isWindows && process.getuid() === 0)) {
+    const err = /TypeError: "uid" must be an integer/;
+
+    pass('uid', undefined);
+    pass('uid', null);
+    pass('uid', process.getuid());
+    fail('uid', __dirname, err);
+    fail('uid', true, err);
+    fail('uid', false, err);
+    fail('uid', [], err);
+    fail('uid', {}, err);
```

Comment:
I suggest adding non-integral values to the tests:

```
> child_process.spawnSync("cat", {uid: 3.5})
node: ../deps/uv/src/unix/core.c:166: uv_close: Assertion `0' failed.
zsh: abort (core dumped)  ./node
% ./node --version
v8.0.0-pre
```

^--- I'm not sure if I should report as a bug, if you type-check for non-integral values, then whatever is broken in uv won't be reachable from node.